sigh/lib/sigh/commands_generator.rb

```diff
@@ -125,7 +125,7 @@ def run
         c.description = 'Manage installed provisioning profiles on your system.'
 
         c.option('-f', '--force', 'Force remove all expired provisioning profiles. Required on CI.')
-        c.option('-e', '--clean_expired', 'Remove all expired provisioning profiles.')
+        global_option('-e', '--clean_expired', 'Remove all expired provisioning profiles.')
```

Comment:
I don’t think we want to do this hear. The issue isn’t with defining this with `c.option`. The issue is actually in at https://github.com/fastlane/fastlane/blob/e874a47c6e2e0e61590a03d3b71e75e5a505d1ce/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb#L50

If you comment this line out, `-e` works totally fine. This might have been an artifact/regression from https://github.com/fastlane/fastlane/pull/18599

I don’t FULLY understand what `parse_global_options` - https://github.com/commander-rb/commander/blob/203dae340d92f1d116a2c6cb55acf310fc077b3a/lib/commander/runner.rb#L372-L385

It seems like `OptionParser.parse! changed behavior in Ruby 2.6 because `-e` works in Ruby 2.5

But… `-e` does work in Ruby 2.6 if you also pass `--verbose` along with it which is super weird!

A solution I found that seems to work is to add a begin/rescue around `parse_global_options` like…

```ruby
begin
  parse_global_options
rescue
  # Don’t need to rescue but lets put a better comment here
end
```

cc: @crazymanish @minuscorp @ainame

Reply:
This is actually super weird, and got broken so randomly

Reply:
I have nooooooo idea. But it doesn’t raise an error if `--verbose` is used which is the most confusing part. I went as far as I could go before having to jump into the Ruby source 😬

Reply:
I am going to try Josh's suggestion now, 

> Why is actually raising an error in that call?🤔

Looks like Ruby's [OptionParser](https://ruby-doc.org/stdlib-2.7.1/libdoc/optparse/rdoc/OptionParser.html) has some known bugs 🥺
Source: https://github.com/commander-rb/commander/issues/6#issuecomment-88361499

Reply:
Hi @joshdholtz @minuscorp 
Look what I found, If I am changing `-e` to `-b` ...There is no exception, all is well...so it seemed **only `-e` option is causing an issue 🤷‍♂️** 

<img width="1513" alt="Screenshot 2021-05-27 at 20 45 28" src="https://user-images.githubusercontent.com/5364500/119880529-a1443480-bf2c-11eb-8a11-9439a246b6a7.png">

Reply:
I think it is colliding with some `-e` global option some level up in the command stack. Is the only thing that come to my mind.

Reply:
Hi @joshdholtz @minuscorp 
Requesting you to re-review this PR now, please! 

> I have nooooooo idea. But it doesn’t raise an error if --verbose is used which is the most confusing part. 

Appending `--verbose` for `fastlane sigh manage -e` command works great! 😇

Reply:
I don't think this is a wanted behavior, even if it patches the issue. It will trigger with any `-e` string, the long and the short form for example. If we cannot track the underlying issue we might need to take another approach😞

Reply:
Yes, I agree!

I just tested the below commands for this patch and everything seems to works great! 😅
- `f sigh manage -e` 
- `f sigh manage -e  --verbose` 
- `f sigh manage -e -f --verbose` 
- `f sigh manage -e -f --verbose  --verbose` 

I just saw, Josh has cloned the Ruby/OptionParser git repo, hopefully, we will found the root cause 🚀 💪